llumnix/arg_utils.py

```diff
@@ -360,6 +368,100 @@ class LaunchArgs:
     launch_mode: LaunchMode = None
     backend_type: BackendType = None
 
+class LlumnixEngineArgs(ABC):
+
+    def __init__(
+        self, engine_args, backend_type: BackendType
+    ) -> None:
+        self.engine_args = engine_args
+        self.override_engine_args = None
+        self.backend_type: BackendType = backend_type
+
+    @abstractmethod
+    def get_overridden_engine_args(self):
+        # returun the engine args after overriding
+        pass
+
+    @abstractmethod
+    def get_engine_world_size(self):
+        pass
+
+    def update_arg(self, args_key: str, args_value):
+        if self.override_engine_args and hasattr(self.override_engine_args, args_key):
+            setattr(self.override_engine_args, args_key, args_value)
+
+    def update_args(self, **kwargs):
+        for args_key, args_value in kwargs.items():
+            self.update_arg(args_key, args_value)
+
+
+class LlumnixEngineArgsFactory:
+
+    def __init__(
+        self,
+        load_registered_service: bool,
+        enable_port_increment: bool,
+        load_registered_service_path: str,
+        pdd_config: PDDConfig,
+    ) -> None:
+        self.load_registered_service: bool = load_registered_service
+        self.load_registered_service_path: str = load_registered_service_path
+        self.pdd_config: PDDConfig = pdd_config
+        self.engine_args_dict: dict[str, LlumnixEngineArgs] = {}
+        self.enable_port_increment: bool = enable_port_increment
+        self.disagg_options_token_port_offset = 0  # used in bladellm
+
+        if self.load_registered_service:
+            if (
+                not self.pdd_config.enable_pd_disagg
+                and not self.pdd_config.enable_engine_pd_disagg
+            ):
+                instance_type_list = ["no_constraints"]
+            else:
+                instance_type_list = ["prefill", "decode"]
+            for instance_type in instance_type_list:
+                self.engine_args_dict[instance_type] = load_engine_args(
+                    instance_type, self.load_registered_service_path
+                )
+
+    def gen_next_engine_args(
```

Comment:
I think gen_next_engine_args should be abstract method too. And therefore can remove the isinstance codes.